source_scraping.py

```diff
@@ -0,0 +1,977 @@
+        - `end_date` 
+            datetime object, the date of the latest news release to be retrieved. By default, this is set to the current date
+        - `verbose`
+            boolean, whether or not the function should print updates
+
+    Returns: a DataFrame containing news releases from the government of Quebec.
+    """
+
+    if start_date > end_date:
+        if verbose: print("Cannot search between {} and {}".format(start_date, end_date))
+        return pd.DataFrame([], columns=_columns)
+
+    
+    region = 'Quebec'
+    sub_region = ''
+    
+    if verbose: print("\nLoading {} Releases between {} and {}\n".format(region, start_date.strftime('%B %d, %Y'), end_date.strftime('%B %d, %Y')))
+    
+    url_base = "http://www.fil-information.gouv.qc.ca/Pages/Articles.aspx?lang=en&Page="
+    page = 1
+    
+    rows = []
+    
+    while True:
+        url = url_base + str(page)
+        
+        if verbose: print("Searching page {}".format(page))
+        
+        response = requests.get(url)
+        soup = BeautifulSoup(response.content, 'html.parser')
+                
+        sections = soup.find_all('section', {"id" : "articles"})
+        
+        for section in sections:
+            date_str = section.time['datetime']
+            ar_date = datetime.strptime(date_str, "%Y-%m-%d")
+                    
+            if ar_date < start_date:
+                if verbose: print("Stopping search at date {}".format(ar_date))
+                return pd.DataFrame(rows, columns=_columns)
+            
+            if ar_date > end_date: # Articles that follow the `end_date` parameter are ignored
+                continue
+
+            
+            for a in section.find_all('a'):
+            
+                link = a['href']
+                title = a.text.replace('\r', '')
+                title = title.replace('\n', '')
+
+                body_soup = BeautifulSoup(requests.get(link).content, 'html.parser')
+                body = body_soup.find('div', class_="article").text
+
+                row = [ar_date, _country, region, sub_region, link, _src_cat, title, body]
+                rows.append(row)
+            
+        page += 1
+
+def _load_newfoundland(start_date=datetime(2020, 1, 1), end_date=datetime.today(), verbose=True):
+    """
+    Parameters: 
+        - `start_date` 
+            datetime object, the date of the earliest news release to be retrieved. By default, only the releases published before Jan 1 2020 are retrieved
+        - `end_date` 
+            datetime object, the date of the latest news release to be retrieved. By default, this is set to the current date
+        - `verbose`
+            boolean, whether or not the function should print updates
+
+    Returns: a DataFrame containing news releases from the government of Newfoundland.
+    """
+    
+    if start_date > end_date:
```

Comment:
It might be easier to move the block into the `_load_province()` function instead, as `load_function()` chops up the provided interval and passes each chunk to `_load_province()`. Would that be alright?

Reply:
Good idea!